CMakeLists.txt

```diff
@@ -163,5 +166,22 @@ if(SOCI_TESTS)
   add_subdirectory(tests)
 endif()
 
+###############################################################################
+# build config file
+###############################################################################
```

Comment:
Cool idea to build config from any SOCI_HAVE variable defined in scope, but I prefer be more explicit in my intentions.

So, IMO, soci-config.h.in should contains all explicit variables which should be defined using either
```#cmakedefine SOCI_HAVE_VAR ${SOCI_HAVE_VAR}```
or
```#cmakedefine SOCI_HAVE_VAR @SOCI_HAVE_VAR@```

which allows to document any defines in template and in result file.

@mloskot What do you think? I'm not insist on it, but usable for documentation purposes.